i18n-helpers/src/lib.rs

```diff
@@ -603,25 +702,37 @@ pub fn reconstruct_markdown(
 ///     ],
 /// );
 /// ```
-pub fn extract_messages(document: &str) -> Vec<(usize, String)> {
+pub fn extract_messages(document: &str) -> Vec<(usize, ExtractedMessage)> {
     let events = extract_events(document, None);
     let mut messages = Vec::new();
     let mut state = None;
 
     for group in group_events(&events) {
         match group {
-            Group::Translate(events) => {
+            Group::Translate(events, group_extra) => {
                 if let Some((lineno, _)) = events.first() {
                     let (text, new_state) = reconstruct_markdown(&events, state);
                     // Skip empty messages since they are special:
                     // they contains the PO file metadata.
                     if !text.trim().is_empty() {
-                        messages.push((*lineno, text));
+                        messages.push((
+                            *lineno,
+                            ExtractedMessage {
+                                message: text,
+                                translator_comment: {
+                                    if group_extra.translator_comments.is_empty() {
+                                        None
+                                    } else {
+                                        Some(group_extra.translator_comments.join(" "))
+                                    }
+                                },
+                            },
+                        ));
                     }
                     state = Some(new_state);
                 }
             }
-            Group::Skip(events) => {
+            Group::Skip(events, _group_extra) => {
```

Comment:
Just use `_` for unnecessary parts in a pattern:

```suggestion
            Group::Skip(events, _) => {
```

Reply:
Okay, done.